Dynatrace/dynatrace-operator
# Always pull the connection info from remote

## Description

Fixes an issue where the endpoint connection info is never updated during the reconciliation loop, when the OneAgent code modules image option is not used (neither `spec.oneAgent.cloudNativeFullStack.codeModulesImage` nor `spec.oneAgent.applicationMonitoring.codeModulesImage`).

Ticket 121913

## How can this be tested?

1. deploy a dynakube that uses "spec.oneAgent.applicationMonitoring.useCSIDriver: true"
2. restart a workload in a targeted namespace so it is auto-injected
3. change the OneAgent connection endpoint confiiguration (e.g. by deploying an additional AG for the same network zone)
4. restart the same workload from step 2 once more
5. wait for the dynakube instance to be reconciled by the provisioner container (e.g. by checking the provisioner log for a message containing "reconciling DynaKube", usually happens within 3-5 minutes)
6. verify the property "serverAddress" in file /opt/dynatrace/oneagent-paas/agent/conf/ruxitagentproc.conf in the app container's file system contains the updated OneAgent connection endpoints reflecting the change made in step 3

## Checklist
